ros2/examples/fibonacci_bt.cpp

```diff
@@ -0,0 +1,114 @@
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of PickNik Inc. nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+#include "example_interfaces/action/fibonacci.hpp"  // for Fibonacci
+#include "fibonacci_action_client.hpp"
+
+#include <chrono>    // for milliseconds
+#include <iostream>  // for cin.get()
+#include <thread>    // for sleep_for
+
+#include <rclcpp/logger.hpp>  // for rclcpp::Loggers
+#include <rclcpp/rclcpp.hpp>  // for Node, MultiThreadedExecutor, ok()
+
+#include <behaviortree_cpp_v3/bt_factory.h>  // for BehaviorTreeFactory, NodeStatus
+#include <behaviortree_ros2/ros2_action_api.hpp>  // for ActionClientApi
+
+namespace {
+using namespace std::chrono_literals;
+auto const LOGGER = rclcpp::get_logger("BehaviorTree");
+auto const SERVER_RESPONSE_TIME = 100ms;
+constexpr auto TREE_RATE = 10ms;
+constexpr auto BT_DESCRIPTION = R"(
+    <?xml version="1.0"?>
+    <root>
+        <BehaviorTree>
+            <Sequence>
+                <FibonacciAction/>
+            </Sequence>
+        </BehaviorTree>
+    </root>)";
+}  // namespace
+
+int main(int argc, char* argv[]) {
+  // Initialize ROS2
+  rclcpp::init(argc, argv);
+  auto const shared_node =
+      std::make_shared<rclcpp::Node>("example_node", rclcpp::NodeOptions{});
+
+  // Create action API handle
+  using Fibonacci = example_interfaces::action::Fibonacci;
+  auto action_api_handle =
+      std::make_shared<BT::ros2::ActionClientApi<Fibonacci>>(
+          shared_node, "fibonacci", SERVER_RESPONSE_TIME);
+
+  // Create BT
+  BT::BehaviorTreeFactory factory;
+  factory.registerBuilder<BT::ros2::FibonacciClient>(
+      "FibonacciAction",
+      [&](const std::string& name, BT::NodeConfiguration const& config) {
+        return std::make_unique<BT::ros2::FibonacciClient>(name, config,
+                                                           action_api_handle);
+      });
```

Comment:
Is there a way to implement this so that the action topic name (`"fibonacci"` in this example) is not hard-coded when registering the action client node? While exploring porting some of my code to use this template, I found that I wanted to be able to set the action topic name dynamically, and that sort of thing is precluded by how registering the action client node works here.

Reply:
@schornakj Sorry it took me so long to respond. Do you want to have multiple nodes of the same type (e.g. FibonacciClient), in this case I think you need to have multiple ActionClientApi's, one for each action interface, or do you want to change the client's target server at runtime? The latter case is not possible because the ROS2 action client gets created at construction of the ActionClientApi object with `rclcpp_action::create_client ` ([here](https://github.com/sjahr/BehaviorTree.ROS/blob/f0e8ec39a66b3b04c7e158e322f1b54b86a11550/ros2/include/behaviortree_ros2/ros2_action_api.hpp#L78) is the related code) and I am not aware that there is a way to re-target an existing client. 

How did you solve this problem?